File: exercism/haskell/bob/src/Bob.hs
module Bob (responseFor) where

import Data.Char (isAlpha, isSpace, toUpper)
import Data.List (dropWhileEnd)

responseFor :: String -> String
responseFor = reply . trim
  where
    reply s | null s || all isSpace s              = "Fine. Be that way!"
            | any isAlpha s && s == fmap toUpper s = "Whoa, chill out!"
            | last s == '?'                        = "Sure."
            | otherwise                            = "Whatever."
    trim = dropWhileEnd isSpace . dropWhile isSpace

File: exercism/clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn response-for [s]
  (cond
    (str/blank? s) "Fine. Be that way!"
    (and
      (some #(Character/isLetter %) s)
      (= s (str/upper-case s))) "Whoa, chill out!"
    (str/ends-with? s "?") "Sure."
    :else "Whatever."))
